src/modules/jcms_notifications/src/JCMSQueueService.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+namespace Drupal\jcms_notifications;
+
+use Aws\Sqs\SqsClient;
+use Drupal\Core\Site\Settings;
+
+/**
+ * Class JCMSQueueService.
+ *
+ * @package Drupal\jcms_notifications
+ */
+class JCMSQueueService {
+
+  /**
+   * @var \Aws\Sqs\SqsClient
+   */
+  protected $sqsClient;
+
+  protected $endpoint = '';
+
+  protected $queueName = '';
+
+  protected $region = '';
+
+  /**
+   * JCMSQueueService constructor.
+   *
+   * @param \Aws\Sqs\SqsClient $sqs_client
+   */
+  public function __construct(SqsClient $sqs_client = NULL) {
+    $this->endpoint = Settings::get('jcms_sqs_endpoint');
+    $this->queueName = Settings::get('jcms_sqs_queue');
+    $this->region = Settings::get('jcms_sqs_region');
+    $config = [
+      'profile' => 'default',
+      'version' => 'latest',
+      'region' => $this->region,
+    ];
+    if (!empty($this->endpoint)) {
+      $config['endpoint'] = $this->endpoint;
+    }
+    $this->sqsClient = $sqs_client ?: new SqsClient($config);
+  }
+
+  /**
+   * Gets the article data from SQS.
+   *
+   * @return array
+   */
+  public function getArticleData() {
+    $queue = $this->sqsClient->getQueueUrl([
+      'QueueName' => $this->queueName,
+    ]);
+    $messages = [];
+    while (TRUE) {
+      $message = $this->sqsClient->receiveMessage([
+        'QueueUrl' => $queue['QueueUrl'],
+        'VisibilityTimeout' => 60,
+        'WaitTimeSeconds' => 20,
+      ]);
+      $sqs_message = $message->get('Messages');
+      if ($sqs_message !== NULL) {
+        $sqs_message = array_shift($sqs_message);
+        $messages[] = $sqs_message;
+        $this->sqsClient->deleteMessage([
```

Comment:
the `deleteMessage` that acknowledges delivery seems to be located too soon; this call permanently removes the message from the queue, therefore the pattern is:
```
while (true) {
    $message = $sqsClient->receiveMessage();
    // make HTTP calls, insert in database, commit transaction, do whatever you need to do...
    $sqsClient->deleteMessage(...);
```